### PR TITLE
dev: Add event tracking for Light / Dark mode toggle

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-972

## Proposed changes
Adds event tracking of light/dark mode toggle for Matomo

---

## Reviewer notes
The `<ThemeToggle />` component is not yet being displayed in the page header, so in order to review this I'm going to manually deploy this branch to the dev environment, login to Matomo and verify that tracking data is showing up when the light/dark mode toggle is clicked. If you have a CAC, please do the same. After this PR is approved, I'll be sure to hide the `<ThemeToggle />` component again before merging into `main`.

Also, pull this branch down on your local machine and verify that all tests are passing.

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests

### CODE REVIEW

File: src/components/Header/Header.tsx
Comment (gidjin) [inline src/components/Header/Header.tsx:104]: nit-pick: we'll want to remove this before it's merged. I know you know just putting a reminder :)

File: src/components/ThemeToggle/ThemeToggle.tsx
Comment (gidjin) [inline src/components/ThemeToggle/ThemeToggle.tsx:15]: Question: Can this call include what state was chosen? Like `Toggling to Dark mode` & `Toggling to Light mode`?

Comment (jcbcapps) [reply src/components/ThemeToggle/ThemeToggle.tsx:15]: Glad you brought this up. I had initially implemented it that way, but wasn't sure if we wanted that or not? If so, it's an easy change.

Comment (gidjin) [reply src/components/ThemeToggle/ThemeToggle.tsx:15]: Personally I would like to see that information too, but we can bring this up at sync and see what others would like.

**Comment by jbecker01** [comment]
Looks good!

File: src/components/ThemeToggle/ThemeToggle.tsx
Comment (abbyoung) [reply src/components/ThemeToggle/ThemeToggle.tsx:15]: where did we land with this?

**Review by abbyoung — COMMENTED** [review]
Looks good! I know you've got to resolve some conflicts with main -- once that's done I'm ready to ✅

File: src/components/ThemeToggle/ThemeToggle.tsx
Comment (jcbcapps) [reply src/components/ThemeToggle/ThemeToggle.tsx:15]: Seems that we're holding off for now since we'll be introducing additional themes in the future.

**Review by abbyoung — APPROVED** [review]

File: src/components/ThemeToggle/ThemeToggle.tsx
Comment (gidjin) [reply src/components/ThemeToggle/ThemeToggle.tsx]: I think we talked about this at Sync when you were out, John was fine waiting for a more detailed event tracking at a later stage.

**Pull request merged**
